Solutions/Menes.Specs/Steps/TestClasses/PetHalDocumentMapperWithContext.cs

```diff
@@ -0,0 +1,23 @@
+﻿// <copyright file="PetHalDocumentMapperWithContext.cs" company="Endjin">
+// Copyright (c) Endjin. All rights reserved.
+// </copyright>
+
+namespace Menes.Specs.Steps.TestClasses
+{
+    using Menes.Hal;
+
+    public class PetHalDocumentMapperWithContext : IHalDocumentMapper<Pet, MappingContext>
+    {
+        public bool LinkMapConfigured { get; private set; }
```

Comment:
It ever so slightly offends me that well over half of this class is identical to `PetHalDocumentMapper`. My inclination would have been to put the common bits in a common base class. And that's not so much to avoid duplication, as it is to make the meaningful differences stand out.

However I would concede that adding an extra moving part in the form of a base class increases complexity so reasonable people could disagree over whether it is a net win. My main reason for favouring this refactoring is that in practice I've often found that when looking at tests (particularly coming back to them when trying to understand existing functionality prior to augmenting or otherwise modifying it) I find it much easier to work out what's going on if test code focuses on what's distinctive about a particular scenario, instead of buring the one important detail in a cloud of incidental baggage.

Reply:
I did consider a base class but it seemed like that would be equally confusing as now there'd be three things to keep in mind instead of two - to me, the naming was enough to express the difference. How strongly do you feel about this?

Reply:
Not very. If you're inclined not to change it having read my argument, then that's fair enough—there are pros and cons either way.